runtime/jcl/common/com_ibm_oti_vm_VM.c

```diff
@@ -156,3 +156,15 @@ Java_com_ibm_oti_vm_VM_markCurrentThreadAsSystemImpl(JNIEnv *env)
 
 	return rc;
 }
+
+/**
+ * Gets the J9ConstantPool address from a J9Class address
+ * @param j9clazz J9Class address
+ * @return address of J9ConstantPool
+ */
+jlong JNICALL
+Java_com_ibm_oti_vm_VM_getJ9ConstantPoolFromJ9Class(JNIEnv *env, jclass unused, jlong j9clazz)
+{
+	J9Class *clazz = (J9Class *)(IDATA)j9clazz;
+	return *(jlong *)&clazz->ramConstantPool;
```

Comment:
@EricYangIBM @tajila 

This is not reasonable where `sizeof(UDATA) != sizeof(jlong)` (i.e. 32-bit platforms).
I suggest it should be:
```
    return (jlong)(IDATA)clazz->ramConstantPool;
```
By the way, why is that field of `J9Class` declared like this
```
    UDATA *ramConstantPool;
```
and not like this?
```
    J9ConstantPool *ramConstantPool;
```

Reply:
With `*(jlong *)&clazz->ramConstantPool` is the problem that `clazz->ramConstantPool` is 32 bits storage when `*(jlong *)&` accesses 64 bits?

Also with `(jlong)(IDATA)clazz->ramConstantPool` maybe casting from 32 bit IDATA to 64 bit jlong would sign extend when the value is e.g. 0x8xxxxxxx.

Reply:
As Eric said we are trying to avoid sign extension.

Perhaps this better:

```
	J9Class *clazz = (J9Class *)(IDATA)j9clazz;
	U_64 ramCP = (U_64)(UDATA)clazz->ramConstantPool; //copy potential 32bit value to a 64bit container
	return *(jlong *)&ramCP;
```

I can see a issue with `*(jlong *)&clazz->ramConstantPool` if a U_64 load is performed directly on the ramCP address  in 32bit cases.

Reply:
Alternatively, we could also do 
```
#if compressedRefs
return (jlong)(IDATA)clazz->ramConstantPool & 0x00000000FFFFFFFF;
#else
return (jlong)(IDATA)clazz->ramConstantPool;
#endif
```

Reply:
You're right about the desire to zero-extend instead of sign-extend: I'll fix my PR.